File: p_34.c
#include <stdio.h>

/* Find sum of factorians. */

int digit_fac_check(unsigned int n, int factorial[]) {
    unsigned int sum = 0;
    unsigned int n_copy = n;
    for (; n >= 10; n /= 10)
        sum += factorial[n % 10];
    return sum + factorial[n] == n_copy;
}

int main() {
    int sum = 0;
    int factorial[10] = {1, 1, 2, 6, 24, 120, 720, 5040, 40320, 362880};
    for (unsigned int i = 10; i < 100000000; ++i) {
        if (digit_fac_check(i, factorial))
            sum += i;
    }
    printf("%u\n", sum);
}

File: p_34.py
from math import factorial

''' Find sum of factorians. '''

factorial_list = [factorial(n) for n in xrange(10)]

def fac_equals_digits(n):
    fac_sum = 0
    n_copy = n
    while n >= 10:
        fac_sum += factorial_list[n % 10]
        n //= 10
    return fac_sum + factorial_list[n] == n_copy

fac_sum = 0
for i in xrange(10, 100000):
    if fac_equals_digits(i):
        fac_sum += i
print fac_sum
